run_linux.go

```diff
@@ -1003,28 +1003,50 @@ func addRlimits(ulimit []string, g *generate.Generator, defaultUlimits []string)
 		g.AddProcessRlimits("RLIMIT_"+strings.ToUpper(ul.Name), uint64(ul.Hard), uint64(ul.Soft))
 	}
 	if !nofileSet {
-		max := define.RLimitDefaultValue
+		// For nofile, podman first tries to set both the hard and soft limits for the current
+		// process to RLimitDefaultValue - this will be successful in most (but not all)
+		// non-rootless environments. If this fails (e.g. in a rootless environment) it will ensure
+		// that the soft limit for the current process is increased to match the hard limit (see
+		// cmd/podman/early_init_linux.go).
 		var rlimit unix.Rlimit
+		rlimit.Cur = define.RLimitDefaultValue
+		rlimit.Max = define.RLimitDefaultValue
+		err := unix.Setrlimit(unix.RLIMIT_NOFILE, &rlimit)
```

Comment:
If the limits are set higher than this on entry, are we lowering them for the calling process?  I don't know that I want that as a side-effect.

Reply:
If the limits are currently greater than `define.RLimitDefaultValue` then yes, they will be lowered (but for the current `buildah` process, not for any calling process).

There are three reasons why I don't think this is an issue:
* `define.RLimitDefaultValue` is 1048576, which matches the default kernel hard limit;
* the purpose of #5275 (which this PR is a continuation of) was to match the behaviour of `podman` (and this is what `podman` does);
* if a user does need a higher limit for some reason, they can simply set the `--ulimit` option.